bigbluebutton-html5/imports/startup/client/auth.js

```diff
@@ -79,6 +94,8 @@ export function joinRouteHandler(nextState, replace, callback) {
             log('error', `Caught: ${e.message}`);
           }
 
+          makeCall('addUserSettings', meetingID, internalUserID, key, value);
```

Comment:
You are making one call for each item. You could do a single call with all of them and validate if they are valid in the server instead of client.